ap/plotting/plotter.py

```diff
@@ -0,0 +1,181 @@
+    #       buggy error bars and the documentation is clearly wrong (mplhep 0.3.12,
+    #       hist 2.4.0), so adjust the output to make up for that, but maybe update or
+    #       remove the next lines if this is fixed to not correct it "twice"
+    from hist.intervals import poisson_interval
+    yerr = poisson_interval(values, variances)
+    yerr[np.isnan(yerr)] = 0
+    yerr[0] = values - yerr[0]
+    yerr[1] -= values
+
+    defaults = {
+        "x": h.axes[0].centers,
+        "y": values,
+        "yerr": yerr,
+        "color": "k",
+        "linestyle": "none",
+        "marker": "o",
+        "elinewidth": 1,
+    }
+    defaults.update(kwargs)
+    ax.errorbar(**defaults)
+
+
+def plot_all(plot_config, style_config, ratio=True):
+    """
+    plot_config expects dictionaries with fields:
+    "method": str, identical to the name of a function defined above,
+    "hist": hist.Hist or hist.Stack,
+    "kwargs": dict (optional),
+    "ratio_kwargs": dict (optional),
+
+    style_config expects fields (all optional):
+    "ax_cfg": dict,
+    "rax_cfg": dict,
+    "legend_cfg": dict,
+    "CMS_label_cfg": dict,
+    """
+
+    plt.style.use(mplhep.style.CMS)
+
+    rax = None
+    if ratio:
+        fig, (ax, rax) = plt.subplots(2, 1, gridspec_kw=dict(height_ratios=[3, 1], hspace=0), sharex=True)
+    else:
+        fig, ax = plt.subplots()
+
+    for key in plot_config:
+        cfg = plot_config[key]
+        if "method" not in cfg:
+            raise ValueError("No method given in plot_cfg entry {key}")
+        method = cfg["method"]
+
+        if "hist" not in cfg:
+            raise ValueError("No histogram(s) given in plot_cfg entry {key}")
+        hist = cfg["hist"]
+        kwargs = cfg.get("kwargs", {})
+        globals()[method](ax, hist, kwargs)
+
+        if ratio and "ratio_kwargs" in cfg:
+            # take ratio_method if the ratio plot requires a different plotting method
+            method = cfg.get("ratio_method", method)
+            globals()[method](rax, hist, cfg["ratio_kwargs"])
+
+    # axis styling
+    ax_kwargs = {
+        "ylabel": "Counts",
+        "xlabel": "variable",
+        "yscale": "linear",
+    }
+    ax_kwargs.update(style_config.get("ax_cfg", {}))
+    if ax_kwargs["yscale"] == "linear":
+        ax_kwargs["ylim"] = 0.000001
+    ax.set(**ax_kwargs)
+
+    if ratio:
+        # hard-coded line at 1
+        rax.axhline(y=1.0, linestyle="dashed", color="gray")
+        rax_kwargs = {
+            "ylim": (0.75, 1.25),
+            "ylabel": "Ratio",
+            "xlabel": "Variable",
+            "yscale": "linear",
+        }
+        rax_kwargs.update(style_config.get("rax_cfg", {}))
+        rax.set(**rax_kwargs)
+
+    # legend
+    legend_kwargs = {
+        "title": "Processes",
+        "ncol": 1,
+        "loc": "upper right",
+    }
+    legend_kwargs.update(style_config.get("legend_cfg", {}))
+    ax.legend(**legend_kwargs)
+
+    CMS_label_kwargs = {
```

Comment:
We should use all lower-case variable names  :)

Reply:
sure, will do